Neos.Flow/Classes/Mvc/ActionResponse.php

```diff
@@ -264,4 +265,52 @@ public function mergeIntoComponentContext(ComponentContext $componentContext): C
 
         return $componentContext;
     }
+
+    /**
+     * Note this is a special use case method that will apply the internal properties (Content-Type, StatusCode, Location, Set-Cookie and Content)
+     * to the given PSR-7 Response and return a modified response. This is used to merge the ActionResponse properties into a possible HttpResponse
+     * created in a View (see ActionController::renderView()) because those would be overwritten otherwise. Note that any component parameters will
+     * still run through the component chain and will not be propagated here.
+     *
+     * WARNING: Should this ActionResponse contain body content it would replace any content in the given HttpReponse.
+     *
+     * @param ResponseInterface $httpResponse
+     * @return ResponseInterface
+     * @internal
+     */
+    public function applyToHttpResponse(ResponseInterface $httpResponse): ResponseInterface
+    {
+        if ($this->statusCode !== 200) {
+            $httpResponse = $httpResponse
+                ->withStatus($this->statusCode);
+        }
+
+        if ($this->hasContent()) {
+            $httpResponse = $httpResponse->withBody($this->content);
+        }
+
+        if ($this->contentType) {
+            $httpResponse = $httpResponse->withHeader('Content-Type', $this->contentType);
+        }
+
+        if ($this->redirectUri) {
+            $httpResponse = $httpResponse->withHeader('Location', (string)$this->redirectUri);
+        }
+
+        foreach ($this->cookies as $cookie) {
+            $httpResponse = $httpResponse->withAddedHeader('Set-Cookie', (string)$cookie);
+        }
+
+        return $httpResponse;
+    }
+
+    /**
+     * Does this action response have content?
+     *
+     * @return bool
+     */
+    private function hasContent(): bool
+    {
+        return $this->content->getSize() > 0;
```

Comment:
can't `$this->content` be null?

Reply:
I realize it's instantiated in the constructor, but maybe we should still check for null in `setContent()`

Reply:
Puhh, I would rather want to make sure we definitely never have `null` values in an instance.

Reply:
Sorry for the confusion, that's what I meant. And I just realized that with the existing check of 
```php
        if (!$content instanceof StreamInterface) {
            $content = stream_for($content);
        }
```
it should not be possible to null the field via `$this->response->setContent(null)`, so all good :)